src/crimson/os/seastore/cache.h

```diff
@@ -572,6 +572,16 @@ class Cache {
       return ret->wait_io().then([ret] {
 	return seastar::make_ready_future<CachedExtentRef>(ret);
       });
+      return ret->wait_io().then([this, ret]() mutable
+      -> get_extent_by_type_ret {
+        if (ret->is_ptr()) {
+	  return seastar::make_ready_future<CachedExtentRef>(ret);
+        } else {
+          touch_extent(*ret);
```

Comment:
Seems that we should `touch_extent` no matter whether `ret->is_ptr()`, am I right?

Reply:
BTW, this seems to be a duplicated `return` clause

Reply:
> Seems that we should `touch_extent` no matter whether `ret->is_ptr()`, am I right?

I have no idea since the old implementation did not. Should we `touch_extent` when we find it in a transaction? @cyx1231st

Reply:
> I have no idea since the old implementation did not. Should we `touch_extent` when we find it in a transaction? @cyx1231st

Ah, sorry, I didn't notice it's retrieved from the transaction, just ignore this comment, please.